packages/nextjs/services/database/repositories/large-stages.ts

```diff
@@ -12,9 +15,29 @@ export type LargeStageWithMilestones = LargeStage & {
 
 export type Status = (typeof stagesStatusEnum.enumValues)[number];
 
-// Note: not used yet
-export async function createStage(stage: LargeStageInsert) {
-  return await db.insert(largeStages).values(stage).returning({ id: largeStages.id });
+export async function createStage(
+  stage: LargeStageInsertWithMilestones,
+): Promise<{ stageId: number; createdMilestones: number[] }> {
+  return await db.transaction(async tx => {
+    const [createdStage] = await tx.insert(largeStages).values(stage).returning({ id: largeStages.id });
+
+    const createdMilestones = [];
+
+    for (let i = 0; i < stage.milestones.length; i++) {
```

Comment:
nit: As an option we could do something like this to make it more readable and use just one insert (didnt't test it, AI suggest)

```
// Prepare all milestone data with the correct stageId and milestoneNumber
    const milestoneData = stage.milestones.map((milestone, index) => ({
      ...milestone,
      stageId: createdStage.id,
      milestoneNumber: index + 1,
    }));

    // Bulk insert all milestones at once
    const createdMilestoneResults = await tx
      .insert(largeMilestones)
      .values(milestoneData)
      .returning({ id: largeMilestones.id });

    // Extract the IDs of the created milestones
    const createdMilestones = createdMilestoneResults.map(result => result.id);
```

Reply:
Thanks! Changed!